pythonAPIClient/client.py

```diff
@@ -298,21 +297,164 @@ def usage_by_client(self, clientID, owner):
+                timezone_name : str
+                    Time zone name of the camera location.
+                reference_logo : str
+                    Reference logo of the camera.
+                reference_url : str
+                    Reference url of the camera.
+                port : str
+                    (ip_camera) Port to connect to camera.
+                brand : str
+                    (ip_camera) Brand of the camera.
+                model : str
+                    (ip_camera) Model of the camera.
+                image_path : str
+                    (ip_camera) Path to retrieve images from the camera.
+                    if the camera is an ip camera and 'is_active_image' is true,
+                    then it will always have a image_path.
+                    However, image_path can exist even if 'is_active_image'
+                    is false for this ip camera.
+                video_path : str
+                    (ip_camera) Path to retrieve video from the camera.
+                    if the camera is an ip camera and 'is_active_video' is true,
+                    then it will always have a video_path.
+                    However, video_path can exist even if 'is_active_video'
+                    is false for this ip camera.
+
+                Raises
+                ------
+                    AuthenticationError
+                        If the client secret of this client object does not match the clientID.
+                    FormatError
+                        List of invalid attributes.
+                    ResourceConflictError
+                        The legacy_cameraID already exist in the database.
+                    InternalError
+                        If there is an API internal error.
+                    ResourceNotFoundError
+                        If no client app exists with the clientID of this client object.
+
+                Returns
+                -------
+                str
+                    cameraID: the new camera ID for the created camera
+        """
+
+        if self.token is None:
+            self.request_token()
+
+        testCamID = '5ae0ecbd336359291be74c12'
+        url = Client.base_URL + 'cameras/' + testCamID
+
+        local_params = dict(locals())
+
+        local_params['type'] = local_params.pop('camera_type')
+
+        del local_params['self']
+        if camera_type == 'ip':
+            local_params['retrieval'] = {
+                'ip': ip,
+                'port': port,
+                'brand': brand,
+                'model': model,
+                'image_path': image_path,
+                'video_path': video_path
+            }
+        elif camera_type == 'non-ip':
+            local_params['retrieval'] = {
+                'snapshot_url': snapshot_url
+            }
+        elif camera_type == 'stream':
+            local_params['retrieval'] = {
+                'm3u8_url': m3u8_url
+            }
```

Comment:
retrieval sub attributes like image_path or snapshot_url should be removed from dictionary. since they are already included in the retrieval.

Reply:
True